src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2645,12 +2936,8 @@ void processContextElement
   //
   // FIXME P6: Once we allow for ServicePath to be modified, this loop must be looked at.
   //
-  int docs = 0;
-
-  // Used to accumulate error response information, checked at the end
-  bool         attributeAlreadyExistsError = false;
-  std::string  attributeAlreadyExistsList  = "[ ";
 
+  std::vector<BSONObj> results;
   while (moreSafe(cursor))
```

Comment:
In this case, element processing within while() was complex (involving a subsequent operation at mongo, i.e. an update()) so I addopted a two-phase approach:

1. In a first phase, all the results are accumulated in a vector of BSONObj, freeing connection after that.
2. The vector elements are processed in a loop.

I'm not sure if this two-phase approach should be adopted in all the cases... My concern is that the results of the cursor may be very long (in this particular case, they aren't: an update is not supposed to match many entities), so a two-phase apporach could be ineficient (i.e two passes and an the memory footprint of the intermediate BSONObj vector).

Opinions are welcome!